server/api.go

```diff
@@ -32,23 +33,174 @@ type StartMeetingFromAction struct {
-	case "/api/v1/meetings":
-		p.handleStartMeeting(w, r)
-	case "/api/v1/config":
-		p.handleConfig(w, r)
-	case "/jitsi_meet_external_api.js":
-		p.handleExternalAPIjs(w, r)
-	default:
+	p.router.ServeHTTP(w, r)
+}
+
+func (p *Plugin) handleJaaSWindow(w http.ResponseWriter, r *http.Request) {
+	if p.getConfiguration().UseJaaS {
+		if p.isJaaSMeeting(r.URL.Path) {
+			p.handleOpenJaaSMeeting(w, r)
+			return
+		}
+	}
+}
+
+func (p *Plugin) handleJaaSSettings(w http.ResponseWriter, r *http.Request) {
+	if !p.getConfiguration().UseJaaS {
+		mlog.Error("error JaaS requested while disabled")
 		http.NotFound(w, r)
+		return
+	}
+
+	if err := p.getConfiguration().IsValid(); err != nil {
+		mlog.Error("Invalid plugin configuration", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var jaasSettingsFromAction JaaSSettingsFromAction
+
+	bodyData, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		mlog.Debug("Unable to read request body", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err1 := json.NewDecoder(bytes.NewReader(bodyData)).Decode(&jaasSettingsFromAction)
+	if err1 != nil {
+		mlog.Debug("Unable to decode the request content as start meeting request or start meeting action")
+		http.Error(w, "Unable to decode your request", http.StatusBadRequest)
+		return
+	}
+
+	var user *model.User = nil
+	userID := r.Header.Get("Mattermost-User-Id")
+	if userID != "" {
+		// Handle moderator
+		userResult, appErr := p.API.GetUser(userID)
+		if appErr != nil {
+			mlog.Debug("Unable to get the user", mlog.Err(appErr))
+			http.Error(w, "Forbidden", http.StatusForbidden)
+			return
+		}
+		user = userResult
+	}
+
+	jaasSettings, err2 := p.getJaaSSettings(jaasSettingsFromAction.Jwt, jaasSettingsFromAction.Path, user)
+	if err2 != nil {
+		mlog.Error("Error getting JaaSSettings", mlog.Err(err2))
+		http.Error(w, "Invalid JaaS settings", http.StatusBadRequest)
+		return
+	}
+
+	settingsJSON, err := json.Marshal(jaasSettings)
+	if err != nil {
+		mlog.Error("Error marshaling the JaaSSettings to json", mlog.Err(err))
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	_, err = w.Write(settingsJSON)
+	if err != nil {
+		mlog.Warn("Unable to write response body", mlog.String("handler", "handleJaaSSettings"), mlog.Err(err))
+	}
+}
+
+func (p *Plugin) isJaaSMeeting(path string) bool {
+	return p.jaasURLCheckRegExp.MatchString(path)
+}
+
+func (p *Plugin) handleJaaSBundle(w http.ResponseWriter, r *http.Request) {
+	if !p.getConfiguration().UseJaaS {
+		http.Error(w, "Not found", http.StatusFound)
+		return
+	}
+
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		mlog.Error("Failed to get the bundle path")
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+
+	jaasMainPath := filepath.Join(bundlePath, "webapp", "dist", "jaas", "jaas-main.js")
```

Comment:
what does Main signify
```suggestion
	jaasPath := filepath.Join(bundlePath, "webapp", "dist", "jaas", "jaas-main.js")
```